### PR TITLE
Added support for reference models with a relative uri

### PR DESCRIPTION
**Pull request by Tiberriver256**
Wound up being a pretty simple change. Let me know if there's anything you'd like to change or add.

Fixes: #182

### CODE REVIEW

**Comment by dnfadmin** [comment]
[CLA assistant check](https://cla.dotnetfoundation.org/OData/ODataConnectedService?pullRequest=183) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.dotnetfoundation.org/OData/ODataConnectedService?pullRequest=183) before we can accept your contribution.<br/><br/>:x: Tiberriver256 [sign now](https://cla.dotnetfoundation.org/OData/ODataConnectedService?pullRequest=183)<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.dotnetfoundation.org/check/OData/ODataConnectedService?pullRequest=183) it.</sub>

**Comment by marabooy** [comment]
@Tiberriver256  could you kindly update the t4 template with these changes. the added changes will be overwritten when the code is regenerated. The changes need to go in the ttinclude file and will probably be around here https://github.com/OData/ODataConnectedService/blob/master/src/Templates/ODataT4CodeGenerator.ttinclude#L877

**Comment by Tiberriver256** [comment]
Anything else funky you can spot @marabooy?

**Comment by marabooy** [comment]
@Tiberriver256 LGTM :) could you add a test case for this code addition.

File: src/Templates/ODataT4CodeGenerator.ttinclude [hunk 1]
```diff
@@ -880,6 +887,12 @@ public class CodeGenerationContext
         {
             return (uri) =>
             {
+                Uri originalUri = uri;
+                if(uri.IsAbsoluteUri == false)
```
Comment (habbes) [inline src/Templates/ODataT4CodeGenerator.ttinclude:891]:
```suggestion
                if(!uri.IsAbsoluteUri)
```

File: src/Templates/ODataT4CodeGenerator.cs [hunk 2]
```diff
@@ -1026,6 +1033,12 @@ internal static IHttpRequestCreator RequestCreator
         {
             return (uri) =>
             {
+                Uri originalUri = uri;
+                if(uri.IsAbsoluteUri == false)
```
Comment (habbes) [inline src/Templates/ODataT4CodeGenerator.cs:1037]:
```suggestion
                if(!uri.IsAbsoluteUri)
```

**Comment by marabooy** [comment]
@Tiberriver256 I would be glad to approve this if there are test cases added.

**Comment by Tiberriver256** [comment]
Sounds good. I'll get it in this weekend.

**Comment by Tiberriver256** [comment]
@marabooy - Test is in. Let me know what you think.

**Review by KenitoInc — APPROVED** [review]
LGTM

**Review by marabooy — APPROVED** [review]
LGTM

**Pull request merged**